src/group_theory/abelianization.lean

```diff
@@ -211,3 +212,68 @@ end abelianization_congr
   left_inv  := λ a, rfl,
   right_inv := by { rintros ⟨a⟩, refl, },
   .. abelianization.of }
+
+section commutator_representatives
+
+open subgroup
+
+/-- Representatives `(g₁, g₂) : G × G` of commutator_set `⁅g₁, g₂⁆ ∈ G`. -/
+def commutator_representatives : set (G × G) :=
+set.range (λ g : commutator_set G, (classical.some g.2, classical.some (classical.some_spec g.2)))
+
+instance [finite (commutator_set G)] : finite (commutator_representatives G) :=
+set.finite_coe_iff.mpr (set.finite_range _)
+
+/-- Subgroup generated by representatives `g₁ g₂ : G` of commutators `⁅g₁, g₂⁆ ∈ G`. -/
+def closure_commutator_representatives : subgroup G :=
+closure (prod.fst '' commutator_representatives G ∪ prod.snd '' commutator_representatives G)
+
+instance closure_commutator_representatives_fg [finite (commutator_set G)] :
+  group.fg (closure_commutator_representatives G) :=
+group.closure_finite_fg _
+
+lemma rank_closure_commutator_representations_le [finite (commutator_set G)] :
+  group.rank (closure_commutator_representatives G) ≤
+    2 * nat.card (commutator_set G) :=
+begin
+  rw two_mul,
+  exact (subgroup.rank_closure_finite_le_nat_card _).trans ((set.card_union_le _ _).trans
+    (add_le_add ((finite.card_image_le _).trans (finite.card_range_le _))
+    ((finite.card_image_le _).trans (finite.card_range_le _ )))),
+end
+
+lemma image_commutator_set_closure_commutator_representatives :
+  (closure_commutator_representatives G).subtype ''
+    (commutator_set (closure_commutator_representatives G)) = commutator_set G :=
+begin
+  apply set.subset.antisymm,
+  { rintros - ⟨-, ⟨g₁, g₂, rfl⟩, rfl⟩,
+    exact ⟨g₁, g₂, rfl⟩ },
+  { exact λ g hg, ⟨_, ⟨⟨_, subset_closure (or.inl ⟨_, ⟨⟨g, hg⟩, rfl⟩, rfl⟩)⟩, ⟨_, subset_closure
+      (or.inr ⟨_, ⟨⟨g, hg⟩, rfl⟩, rfl⟩)⟩, rfl⟩, classical.some_spec (classical.some_spec hg)⟩ },
```

Comment:
In fact, I think these lines are complicated enough that they deserve some more line wrapping, e.g.:
```suggestion
  { exact λ g hg, ⟨_,
      ⟨⟨_, subset_closure (or.inl ⟨_, ⟨⟨g, hg⟩, rfl⟩, rfl⟩)⟩,
       ⟨_, subset_closure (or.inr ⟨_, ⟨⟨g, hg⟩, rfl⟩, rfl⟩)⟩,
       rfl⟩,
      hg.some_spec.some_spec⟩ },
```